libs/governance/src/lib/proposals-hooks/use-proposal-submit.ts

```diff
@@ -0,0 +1,58 @@
+import { useCallback, useState } from 'react';
+import * as Sentry from '@sentry/react';
+import { useVegaWallet, useVegaTransaction } from '@vegaprotocol/wallet';
+import { determineId } from '@vegaprotocol/react-helpers';
+import { useProposalEvent } from './use-proposal-event';
+import type { ProposalSubmission } from '@vegaprotocol/wallet';
+import type { ProposalEvent_busEvents_event_Proposal } from './__generated__/ProposalEvent';
+
+export const useProposalSubmit = () => {
+  const { keypair } = useVegaWallet();
+  const waitForProposalEvent = useProposalEvent();
+
+  const { send, transaction, setComplete, TransactionDialog } =
```

Comment:
vent (non-blocking): for me it feels very, very strange for a pattern to couple hooks with the UI. i'd expect hooks to return some sort of state or action from which you can easily compute what you want to display on the ui.

Reply:
cc @mattrussell36 who was a big fan of this. I quite like it myself, as it allows you to either use default UI or take control of it. What do you see as the downsides?

Reply:
As discussed on slack :-), the hook does now return the dialogOpen state so this could easily be refactored to not return the components as well. Lets consider this in a different PR though